### PR TITLE
PROTO-2985: fix-reputer-staking-events-insertions

### PR DESCRIPTION
**Pull request by KjetilVaa**
## Purpose of Changes and their Description
- This changes processing of reputer staking data from events to transactions
- Expands the schema of reputer_stakes table to include delegator address where that is in the message

## Are these changes tested and documented?
- Tested locally and validated staking data coming in the reputer_stakes table

- [ ] If tested, please describe how. If not, why tests are not needed.
- [ ] If documented, please describe where. If not, describe why docs are not needed.
- [ ] Added to `Unreleased` section of `CHANGELOG.md`?

## Still Left Todo

*Fill this out if this is a Draft PR so others can help.*

### CODE REVIEW

**Review by fernandofcampos — APPROVED** [review]

**Comment by kpeluso** [comment]
@KjetilVaa Your PR description could use way more context as this is now an open source repo.

File: process_tx.go [hunk 1]
```diff
@@ -584,3 +656,40 @@ func insertMsgSend(height uint64, messageId uint64, msg types.MsgSend) error {
...
+			delegator_address,
+			height
+		) VALUES ($1, $2, $3, $4, $5, $6, $7)`,
+		requestType,   // type (text)
+		topicIdInt,    // topic_id (integer)
+		sender,        // sender (text)
+		amountNull,    // amount (numeric, nullable)
+		reputerNull,   // reputer_address (text, nullable)
+		delegatorNull, // delegator_address (text, nullable)
+		height,        // height (integer)
```
Comment (kpeluso) [inline process_tx.go:688]: a bit messy how we have this pseudo-monolithic db.go to handle all db client interactions but then we do some db things closer to the application logic

**Review by kpeluso — COMMENTED** [review]
Some notes on style.

Side note / possible follow up PR (not necessarily for this one): I believe @cal-hawk would prefer to have a table of "current stakes". I believe your PR here is a stepping stone toward that, regardless, just noting the original ask as I understand it.

File: process_tx.go (hunk 1 above)
Comment (KjetilVaa) [reply process_tx.go:688]: Yeah, I agree. I think a refactor is needed. A good time for this is probably when we switch this to consume from Kafka vs RPC.

**Pull request merged**